Instruments/source/CacheTupleProducer.cxx

```diff
@@ -62,17 +63,23 @@ class CacheTupleProducer {
                    % args.input_file() % args.output_file() % args.selections();
 
         auto originalFile = root_ext::OpenRootFile(args.input_file());
+        size_t n_tot_events = 0;
+        std::map<std::string,std::shared_ptr<ntuple::EventTuple>> map_event;
         for(unsigned c = 0; c < channels.size(); ++c){
             auto originalTuple = ntuple::CreateEventTuple(channels.at(c),originalFile.get(),true,ntuple::TreeState::Full);
+            map_event[channels.at(c)] = originalTuple;
+            n_tot_events += originalTuple->GetEntries();
+        }
+        for(unsigned c = 0; c < channels.size(); ++c){
             CacheTuple cache(channels.at(c), outputFile.get(), false);
-            const Long64_t n_entries = std::min(args.maxEvents(),originalTuple->GetEntries());
+            const Long64_t n_entries = std::min(args.maxEvents(),map_event.at(channels.at(c))->GetEntries());
```

Comment:
- don't access map each time. create `const auto& originalTuple = *map_event.at(channels.at(c)).first;` at the beginning on the for and use it.
- for n_entries: `const Long64_t n_entries = map_event.at(channels.at(c)).second;`